Source/scrollrt.cpp

```diff
@@ -916,6 +918,139 @@ void DrawDungeon(const Surface &out, Point tilePosition, Point targetBufferPosit
 	}
 }
 
+Point CheckPredefinedPattern(Point tilePosition)
+{
+	Point pos = { 0, 0 };
+	if (tilePosition.y == -1 && tilePosition.x >= 52 && tilePosition.x <= 61) {
+		// top side sand patch
+		if (tilePosition.x == 61) {
+			// right corner
+			pos = { 64, 66 };
+		} else if (tilePosition.x == 52) {
+			// left corner
+			pos = { 57, 66 };
+		} else {
+			// edge
+			pos = { 61, 66 };
+		}
+	} else if (tilePosition.x == 96 && ((tilePosition.y >= 52 && tilePosition.y <= 71) || (tilePosition.y >= 79 && tilePosition.y <= 89))) {
+		// right side sand patch
+		if (tilePosition.y == 89 || tilePosition.y == 71) {
+			// bottom corner
+			pos = { 65, 74 };
+		} else if (tilePosition.y == 79 || tilePosition.y == 52) {
+			// top corner
+			pos = { 65, 67 };
+		} else {
+			// edge
+			pos = { 65, 72 };
+		}
+	} else if ((tilePosition.y < 0 && IsAnyOf(tilePosition.x, 2, 62, 82))
+	    || (tilePosition.y > 95 && IsAnyOf(tilePosition.x, 32, 84))) {
+		// left river bank
+		pos = { 68, 22 - abs(tilePosition.y % 3) };
+	} else if ((tilePosition.y < 0 && IsAnyOf(tilePosition.x, 3, 63, 83))
+	    || (tilePosition.y > 95 && IsAnyOf(tilePosition.x, 33, 85))) {
+		// right river bank
+		pos = { 69, 22 - abs(tilePosition.y % 3) };
+	}
+	return pos;
+}
+
+Point FillPattern(Point tilePosition)
+{
+	constexpr int PATTERNWIDTH = 5;
+	constexpr int PATTERNHEIGHT = 5;
+	constexpr std::array<Point, 6> patterns { { { 51, 27 }, { 23, 57 }, { 54, 34 }, { 79, 46 }, { 41, 32 }, { 52, 21 } } };
+
+	static int patternRandomizer[MAXDUNX][MAXDUNY] {};
+	static bool doInit = true;
+
+	if (doInit) {
+		doInit = false;
+		std::mt19937 BetterRng;
+		BetterRng.seed(666);
+		std::uniform_int_distribution<int32_t> dist(0, INT_MAX);
+		for (int x = 0; x < MAXDUNX; x++) {
+			for (int y = 0; y < MAXDUNY; y++) {
+				patternRandomizer[x][y] = dist(BetterRng) % patterns.size();

```

Comment:
Can use the distribution to get a value in the expected range directly instead of needing to use mod division. This also removes the need to `#import <climits>`
```suggestion
		std::uniform_int_distribution<int> dist(0, patterns.size() - 1);
		for (int x = 0; x < MAXDUNX; x++) {
			for (int y = 0; y < MAXDUNY; y++) {
				patternRandomizer[x][y] = dist(BetterRng);
```